### ISSUE TITLE
test_cone_vec_repr sometimes fails

### ISSUE THREAD

**Issue by D1rk123** [ahendriksen/tomosipo#11]
test_cone_vec_repr includes the following test:
- generate a random cone_vector geometry
- represent it as a string using the repr function
- recreate a geometry from this string
- check if the original and the recreated geometries are equal

Why this test sometimes fails:
The repr function prints the arrays using the numpy repr function according to the settings in tomosipo.utils.print_options. This results in a precision of 8 digits after the comma in scientific notation. The _eq_ function in a cone_vec geometry tests that every value in the array has a smaller absolute difference than ts.epsilon(1e-8). For values larger than 10 the printing can introduce a larger round off error than ts.epsilon. The source detector distance is randomly generated between 0 and 20, so the values in the dtr_pos vector can be higher than 10 causing the test to fail, but only sometimes.

How to fix it:
One way to fix it would be to replace the float printing mode to floatmode="unique" in all repr functions, resulting in longer strings, but no more rounding errors. I think this would fit well with the Python documentation description of the repr function:
"This is typically used for debugging, so it is important that the representation is information-rich and unambiguous."
(https://docs.python.org/3/reference/datamodel.html#object.__repr__)

Yet it also raises the question whether an absolute error threshold is the right choice for checking equality between geometries. To me it seems like a nice alternative to make == check exact values and to introduce a new function is_close similar to numpy.allclose to check with both relative and absolute thresholds. I know this is quite a small detail, so I'm probably overthinking this, but @ahendriksen  what is your view on this?

**Comment by ahendriksen** [comment]
Weird. This test has never been flakey before. 

Exact equality testing is useless for floating point values imho. Slightly different ordering of operations can introduce epsilon differences in the output. With 64bit floats, 1e-8 is already a wide margin (but also more than tight enough for practical applications). 

TLDR: 

I would keep the eq inplementation like it is now. 

I would adjust the repr test so that it becomes non flakey (instead of changing the repr)
